--- a/src/Validator/InputValidator.php
+++ b/src/Validator/InputValidator.php
@@ -118,7 +118,7 @@
         if (null !== $this->defaultTranslator) {
             // @phpstan-ignore-next-line (only for Symfony 4.4)
             $builder
                 ->setTranslator($this->defaultTranslator)
-                ->setTranslationDomain('validators');
+                ->setTranslationDomain('customerValidators');
         }

         return $builder->getValidator();
