--- a/TypeContext/TypeContextFactory.php
+++ b/TypeContext/TypeContextFactory.php
@@ -201,7 +201,7 @@
         }

         $templates = [];
-        foreach ($this->getPhpDocNode($rawDocNode)->getTagsByName('@template') as $tag) {
+        foreach ($this->getPhpDocNode($rawDocNode)->getTagsByName('@template') + $this->getPhpDocNode($rawDocNode)->getTagsByName('@phpstan-template') as $tag) {
             if (!$tag->value instanceof TemplateTagValueNode) {
                 continue;
             }
